polaris-core/src/main/java/org/apache/polaris/core/config/FeatureConfiguration.java

```diff
@@ -100,6 +100,16 @@ public static void enforceFeatureEnabledOrThrow(
           .defaultValue(false)
           .buildFeatureConfiguration();
 
+  public static final FeatureConfiguration<Boolean> ALLOW_NAMESPACE_LOCATION_ESCAPE =
+      PolarisConfiguration.<Boolean>builder()
+          .key("ALLOW_NAMESPACE_LOCATION_ESCAPE")
```

Comment:
Feature flags [constitute](https://polaris.apache.org/in-dev/unreleased/evolution/) our SemVer "API" surface, so I believe we need to formally send a discussion email to the `dev` ML about this flag.

Reply:
Proactively +1 from my side :)

Reply:
+1 for the change as well. If this is more a bug fix, I don't think we even need this flag, the only reason we need the config is that existing behavior can be broken due to the change.

Reply:
We could even mark it `deprecated` in the beginning(this PR).